src/main/java/com/xebialabs/xlrelease/ci/Credential.java

```diff
@@ -155,15 +189,37 @@ public FormValidation doCheckSecondaryProxyUrl(@QueryParameter String secondaryP
             return validateOptionalUrl(secondaryProxyUrl);
         }
 
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Project context) {
+            // TODO: also add requirement on host derived from URL ?
+            List<StandardUsernamePasswordCredentials> creds = lookupCredentials(StandardUsernamePasswordCredentials.class, context,
+                    ACL.SYSTEM,
+                    HTTP_SCHEME, HTTPS_SCHEME);
+
+            return new StandardUsernameListBoxModel().withAll(creds);
+        }
 
         public FormValidation doValidate(@QueryParameter String xlReleaseServerUrl, @QueryParameter String xlReleaseClientProxyUrl, @QueryParameter String username,
-                                         @QueryParameter Secret password, @QueryParameter String secondaryServerUrl, @QueryParameter String secondaryProxyUrl) throws IOException {
+                                         @QueryParameter Secret password, @QueryParameter String secondaryServerUrl, @QueryParameter String secondaryProxyUrl, @QueryParameter boolean useGlobalCredential, @QueryParameter String credentialsId) throws IOException {
             try {
                 String serverUrl = Strings.isNullOrEmpty(secondaryServerUrl) ? xlReleaseServerUrl : secondaryServerUrl;
                 String proxyUrl = Strings.isNullOrEmpty(secondaryProxyUrl) ? xlReleaseClientProxyUrl : secondaryProxyUrl;
 
+                if (useGlobalCredential && Strings.isNullOrEmpty(credentialsId)) {
+                    return FormValidation.error("No credentials specified");
+                }
+                StandardUsernamePasswordCredentials credentials = lookupSystemCredentials(credentialsId);
+                if (useGlobalCredential && credentials == null) {
+                    return FormValidation.error(String.format("Could not find credential with id '%s'", credentialsId));
+                }
+                if (serverUrl == null || serverUrl.isEmpty()) {
```

Comment:
I just said to use isNullOrEmpty because just 3 lines before is used :)) but agree on your comment